src/lib/vasi-macro/src/lib.rs

```diff
@@ -110,8 +110,31 @@ use syn::Field;
 /// }
 /// ```
 ///
-/// TODO: Extend to support trait bounds. See e.g.
-/// <https://github.com/dtolnay/syn/blob/master/examples/heapsize/heapsize_derive/src/lib.rs#L16>
+/// A generic type *conditionally* implements VirtualAddressSpaceIndependent,
+/// if its type parameters do (as the derive macros in the std crate behave).
+/// ```
+/// use vasi::VirtualAddressSpaceIndependent;
+///
+/// #[derive(VirtualAddressSpaceIndependent)]
+/// struct MyWrapper<T> {
+///   val: T,
+/// }
```

Comment:
Does it work if there are bounds on the generic types? Ex: `struct MyWrapper<T: Debug>` or `struct MyWrapper<T> where T: Debug`. (Not that we need to support that, but maybe leave as a TODO if it doesn't.)

Reply:
Yup these are supported - added another doc test.